gen2-deeplabv3_depth/main.py

```diff
@@ -0,0 +1,167 @@
+detection_nn = pipeline.createNeuralNetwork()
+detection_nn.setBlobPath(nn_path)
+detection_nn.setNumPoolFrames(4)
+detection_nn.input.setBlocking(False)
+detection_nn.setNumInferenceThreads(2)
+
+right = pipeline.createMonoCamera()
+right.setResolution(dai.MonoCameraProperties.SensorResolution.THE_400_P)
+right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+
+manip = pipeline.createImageManip()
+manip.setResize(nn_shape,nn_shape)
+manip.setFrameType(dai.RawImgFrame.Type.BGR888p)
+right.out.link(manip.inputImage)
+manip.out.link(detection_nn.input)
+
+left = pipeline.createMonoCamera()
+left.setResolution(dai.MonoCameraProperties.SensorResolution.THE_400_P)
+left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+
+stereo = pipeline.createStereoDepth()
+stereo.setConfidenceThreshold(200)
+stereo.setOutputDepth(True)
+left.out.link(stereo.left)
+right.out.link(stereo.right)
+
+# Create depth output
+xout_depth = pipeline.createXLinkOut()
+xout_depth.setStreamName("depth")
+stereo.depth.link(xout_depth.input)
+
+# Create output for right frames
+xout_right_cropped = pipeline.createXLinkOut()
+xout_right_cropped.setStreamName("right_cropped")
+detection_nn.passthrough.link(xout_right_cropped.input)
+
+xout_right = pipeline.createXLinkOut()
+xout_right.setStreamName("right")
+right.out.link(xout_right.input)
+
+xout_nn = pipeline.createXLinkOut()
+xout_nn.setStreamName("nn")
+detection_nn.out.link(xout_nn.input)
+
+# Pipeline is defined, now we can connect to the device
+with dai.Device(pipeline) as device:
+    device.startPipeline()
+
+    # Output queues will be used to get the outputs from the device
+    q_cropped = device.getOutputQueue(name="right_cropped", maxSize=4, blocking=False)
+    q_depth = device.getOutputQueue(name="depth", maxSize=4, blocking=False)
+    q_nn = device.getOutputQueue(name="nn", maxSize=4, blocking=False)
+    q_right = device.getOutputQueue(name="right", maxSize=4, blocking=False)
+
+    start_time = time.time()
+    counter = 0
+    fps = 0
+
+    frame = None
+    depth_frame = None
+
+    while True:
+        # instead of get (blocking) used tryGet (nonblocking) which will return the available data or None otherwise
+        in_cropped = q_cropped.tryGet()
+        in_nn = q_nn.tryGet()
+        in_depth = q_depth.tryGet()
+
+        in_right = q_right.tryGet()
+        if in_right is not None:
+            cv2.imshow("right", in_right.getCvFrame())
+
+        if in_cropped is not None:
+            frame = in_cropped.getCvFrame()
+
+        if in_depth is not None:
+            depth_frame = in_depth.getFrame()
+            # Crop the depth map
+            depth_frame = depth_frame[0:400, 120:520]
+            # Resize it so it will match the NN output
+            depth_frame = cv2.resize(depth_frame, (nn_shape,nn_shape))
+
+            # Remove this to disable showing depth frames
+            depthFrameColor = cv2.normalize(depth_frame, None, 255, 0, cv2.NORM_INF, cv2.CV_8UC1)
```

Comment:
```suggestion
            depth_frame = cv2.normalize(depth_frame, None, 255, 0, cv2.NORM_INF, cv2.CV_8UC1)
```
if we would rename this to depth_frame, then in line 161 we wouldn't have to do the same operations again

Reply:
it can be also `depthFrame` (renamed in a whole file) to match the camelCase standard

Reply:
wanted to make it easy for users to add their custom code. I added a function that will color&display depth frames, so this piece of code isn't duplicated